### PR TITLE
Imperia china rework pt2

### PR DESCRIPTION
**Pull request by Emifas**
Holy shit it's finally done GET ME OUT OF HERE

### CODE REVIEW

File: gfx/map/map_object_data/generated_map_object_locators_city.txt
Comment (HiddeLekanne) [inline gfx/map/map_object_data/generated_map_object_locators_city.txt:1]: ehm.... why are we adding this?

File: common/decisions/imperia_qing_decisions.txt [hunk 1]
```diff
@@ -27,4 +27,45 @@
...
+			count >= 5
+		}
+
+		any_scope_state = {
+			OR = {
+			state_region = s:STATE_BEIJING
+			state_region = s:STATE_XIAN
+			state_region = s:STATE_NANJING
+			state_region = s:STATE_GUANGDONG
+			}
```
Comment (Danarca) [inline common/decisions/imperia_qing_decisions.txt]: Formatting

File: common/dna_data/imperia_dna_chi_duan_qirui.txt
Comment (Danarca) [inline common/dna_data/imperia_dna_chi_duan_qirui.txt:1]: For all of the DNA files there's an issue with indentation. `genes` are inside `portrait_info`, but is on the same length, and they closing brackets are innermost..

File: common/character_templates/imperia_china_templates.txt [hunk 2]
```diff
@@ -64,42 +70,838 @@ gansu_viceroy_template = {
...
+}
+
+#Qing Characters
+
+imperia_chi_li_hongzhang_template = { #One of the Four Officials of the Late Qing Dynasty, known contemporarily as the Chinese Bismarck
+	first_name = "Hongzhang"
+	last_name = Li
+	historical = yes
+	culture = cu:han
+    dna = imperia_dna_chi_li_hongzhang
```
Comment (Danarca) [inline common/character_templates/imperia_china_templates.txt]: indentation? Could be spaces instead of tabs, but it looks weird in github.

File: common/character_templates/imperia_china_templates.txt [hunk 3]
```diff
@@ -64,42 +70,838 @@ gansu_viceroy_template = {
...
+	ideology = imperia_ideology_modernizer
+	interest_group = ig_landowners
+	birth_date = 1823.2.15
+	traits = {
+		expert_political_operator
+		masterful_diplomat
+        master_bureaucrat #most chinese generals in this period were good politicians and terrible military leaders
```
Comment (Danarca) [inline common/character_templates/imperia_china_templates.txt]: Indentation

File: common/character_templates/imperia_china_templates.txt [hunk 4]
```diff
@@ -64,42 +70,838 @@ gansu_viceroy_template = {
...
+    }
+    interest_group_leader_usage = {
+        country_trigger = {
+            c:CHI ?= this
+        }
+        interest_group_trigger = {
+        OR = {
+            is_interest_group_type = ig_industrialists
+            is_interest_group_type = ig_intelligentsia
+        }
```
Comment (Danarca) [inline common/character_templates/imperia_china_templates.txt]: Indentation

File: common/character_templates/imperia_china_templates.txt [hunk 5]
```diff
@@ -64,42 +70,838 @@ gansu_viceroy_template = {
...
+		}
+        interest_group_trigger = {
+            is_marginal = no
+            is_interest_group_type = ig_landowners
+        }   
+		earliest_usage_date = 1875.1.1
+		latest_usage_date = 1905.1.1
+		chance = 75
+	}
+}
```
Comment (Danarca) [inline common/character_templates/imperia_china_templates.txt]: Indentation

File: common/character_templates/imperia_china_templates.txt [hunk 6]
```diff
@@ -64,42 +70,838 @@ gansu_viceroy_template = {
...
+    agitator_usage = {
+		country_trigger = {
+			c:CHI ?= this
+		}
+        interest_group_trigger = {
+            OR = {
+                is_interest_group_type = ig_intelligentsia
+                is_interest_group_type = ig_trade_unions
+            }
+        }
```
Comment (Danarca) [inline common/character_templates/imperia_china_templates.txt]: Indentation (you're using spaces here)

File: common/character_templates/imperia_china_templates.txt [hunk 7]
```diff
@@ -64,42 +70,838 @@ gansu_viceroy_template = {
...
+        interest_group_trigger = {
+            OR = {
+                is_interest_group_type = ig_petty_bourgeoisie
+                is_interest_group_type = ig_intelligentsia
+            }
+        }   
+		earliest_usage_date = 1890.1.1
+		latest_usage_date = 1930.1.1
+		chance = 75
+	}
```
Comment (Danarca) [inline common/character_templates/imperia_china_templates.txt]: Indentation

File: common/character_templates/imperia_china_templates.txt
Comment (Danarca) [inline common/character_templates/imperia_china_templates.txt:1]: This entire file could do with an autoformat, tbh

**Review by Danarca — CHANGES_REQUESTED** [review]
Im like 25% done here. 120k lines, wth.
Please do NOT use an autoformatter on the dna files, it'll do a fucky wucky if you do.

File: common/ideologies/imperia_leader_ideologies.txt
Comment (Danarca) [inline common/ideologies/imperia_leader_ideologies.txt:1]: You should rename this to centralizer, just so we dont get confused further ;p

File: common/history/pops/11_east_asia.txt [hunk 8]
```diff
@@ -379,40 +379,64 @@
...
+				size = 5000
 			}
 		}
 	}
 	s:STATE_NORTHERN_MANCHURIA = {
 		region_state:MCH = {
 			create_pop = {
 				culture = manchu
-				size = 2821200
+				size = 400000
```
Comment (Unintuition-Time) [inline common/history/pops/11_east_asia.txt]: Not to nitpick, but I don't enjoy the roundness of these numbers. Try adding a bit of variation to each of them.

File: common/history/power_blocs/imperia_power_blocs.txt [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+			principle = principle_vassalization_2
+
+			# subjects are automatically part of the bloc
+		}
+		if = {
+			limit = {
+				has_dlc_feature = power_bloc_features
+			}
+			power_bloc = {
+				add_principle = principle_internal_trade_2
```
Comment (Unintuition-Time) [inline common/history/power_blocs/imperia_power_blocs.txt]: This will cause errors and issues, as internal_trade requires the Trade League Power Bloc. Please swap it to market_unification.

File: common/ideologies/imperia_leader_ideologies.txt [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+		law_interventionism = approve
+		law_agrarianism = neutral
+		law_traditionalism = disapprove	
+		law_laissez_faire = disapprove
+		law_industry_banned = strongly_disapprove
+	}
+
+	possible = {
+		# Intentionally left empty
+	}
```
Comment (Unintuition-Time) [inline common/ideologies/imperia_leader_ideologies.txt:32]: Isn't leaving it intentionally empty potentially problematic? Why not just copy the possible from our merge file with vtm?

**Review by Unintuition-Time — DISMISSED** [review]

File: common/journal_entries/imperia_china_opium_je.txt [hunk 11]
```diff
@@ -0,0 +1,200 @@
...
+		NOT = {
+			has_law = law_type:law_free_trade
+		}
+			this.market = {
+				any_trade_route = {
+					goods = g:opium
+					count = 0
+					level > 6
+				}
+			}
```
Comment (Danarca) [inline common/journal_entries/imperia_china_opium_je.txt]: Formatting

File: common/static_modifiers/imperia_qing_modifiers.txt [hunk 12]
```diff
@@ -116,6 +121,13 @@ modifier_successful_sinicization = {
 	building_throughput_add = 0.05
 }
 
+modifier_opium_wars_lost_substate = {
+	icon = gfx/interface/icons/timed_modifier_icons/modifier_fire_negative.dds
+    state_turmoil_effects_mult = 0.25
+    country_prestige_mult = -0.25
+    state_standard_of_living_add = -1	
+}
+
```
Comment (Danarca) [inline common/static_modifiers/imperia_qing_modifiers.txt]: Formatting

File: events/imperia_qing_monarchy.txt
Comment (Danarca) [inline events/imperia_qing_monarchy.txt:1]: Something I realised we could do, is set characters as immortal, meaning they wont die before its their (scripted) time.
``` 
## set_character_immortal
Set scoped character as immortal.  
scope:larry = { set_character_immortal = yes/no }  
**Supported Scopes**: character  ```
Dont know if its relevant for this file?

File: events/imperia_qing_substate_events.txt
Comment (Danarca) [inline events/imperia_qing_substate_events.txt:1]: Format this file, please xp

File: events/warlord_china_events.txt [hunk 13]
```diff
@@ -27,14 +27,37 @@ warlord_china_events.100 = { #Split china up into warlord states
...
+				}
+			remove_modifier = modifier_local_armies
+		}
+
+		every_country = {
+			limit = {
+					has_modifier = modifier_chinese_diglossia
+				}
+			remove_modifier = modifier_chinese_diglossia
+		}
```
Comment (Danarca) [inline events/warlord_china_events.txt]: Format

File: events/warlord_china_events.txt [hunk 14]
```diff
@@ -559,16 +588,26 @@ warlord_china_events.100 = { #Split china up into warlord states
...
 			}
 		}
 		every_subject_or_below = {
 			make_independent = yes
 		}
-		ig:ig_armed_forces = {
-			set_interest_group_name = ig_zhili_clique
-		}
 
+		play_as = c:ZHI
```
Comment (Danarca) [inline events/warlord_china_events.txt:618]: Just to be sure, you've tested you dont get a game over when you press this button?

File: events/warlord_china_events.txt [hunk 15]
```diff
@@ -790,3 +829,174 @@ warlord_china_events.301 = {
...
+			value = 0.1
+			culture = cu:han
+		}
+	}
+
+	option = { #People's Republic of China
+		name = warlord_china_events.302.g
+		trigger = {
+			ROOT = {
+					has_law = law_type:law_council_republic
```
Comment (Danarca) [inline events/warlord_china_events.txt]: Format

**Review by Danarca — CHANGES_REQUESTED** [review]
Alright, I've gone most of the relevant files through, and it seems well made besides a few formatting errors ;p

I do question why you included non-Chinese state_region formatting, but oh well :p 

Let's get you out of China asap :D

File: events/warlord_china_events.txt (hunk 14 above)
Comment (Emifas) [reply events/warlord_china_events.txt:618]: Yes, it does indeed tag switch correctly

File: common/ideologies/imperia_leader_ideologies.txt (hunk 10 above)
Comment (Emifas) [reply common/ideologies/imperia_leader_ideologies.txt:32]: @Danarca can you weigh in on this

File: events/imperia_qing_monarchy.txt
Comment (Emifas) [reply events/imperia_qing_monarchy.txt:1]: you can take a look and tell me if i did it correctly :P

File: common/ideologies/imperia_leader_ideologies.txt (hunk 10 above)
Comment (Danarca) [reply common/ideologies/imperia_leader_ideologies.txt:32]: I... _Think_ leaving it empty will allow it to pop up everywhere. If you want to use it only in scripted situations, I'd try giving it an `always = no`

File: events/imperia_qing_monarchy.txt
Comment (Danarca) [reply events/imperia_qing_monarchy.txt:1]: I've never used it myself, but that looks to be the proper way to go about it :)

File: gfx/map/map_object_data/generated_map_object_locators_city.txt
Comment (Emifas) [reply gfx/map/map_object_data/generated_map_object_locators_city.txt:1]: needed for the hubs in new states, i think i explained this last mod meeting

File: tools/mapeditor/map_editor_status.txt
Comment (Danarca) [inline tools/mapeditor/map_editor_status.txt:1]: Was this file needed?

Comment (Emifas) [reply tools/mapeditor/map_editor_status.txt:1]: probably not? i don't think it does anything

File: gfx/map/spline_network/spline_network.splnet
Comment (HiddeLekanne) [inline gfx/map/spline_network/spline_network.splnet:1]: I though you said this didn't work?

Comment (Emifas) [reply gfx/map/spline_network/spline_network.splnet:1]: creating roads between hubs doesn't work due to a bug in the map editor but the hubs are still under spline network. generated map objects contains the locators (telling the game where to put the graphics) but the hubs themselves are not stored there

**Review by HiddeLekanne — APPROVED** [review]

**Review by Danarca — APPROVED** [review]
